setuptools_rust/extension.py

```diff
@@ -240,7 +240,23 @@ def _metadata(self) -> "_CargoMetadata":
             ]
             if self.cargo_manifest_args:
                 metadata_command.extend(self.cargo_manifest_args)
-            self._cargo_metadata = json.loads(subprocess.check_output(metadata_command))
+
+            try:
+                payload = subprocess.check_output(metadata_command, encoding="latin-1")
```

Comment:
Ah, I'm really sorry to have missed on the first read, I think we might need to opt-in to capturing stderr:

```suggestion
                payload = subprocess.check_output(metadata_command, encoding="latin-1", stderr=subprocess.PIPE)
```

Reply:
(could also use `stderr=subprocess.STDOUT` to merge the two together so that the error could just use `e.stdout`, I think both merged and separate have advantages and disadvantages so happy to go with whichever you prefer)

Reply:
Good point, thanks!

I preferred the separate versions (no reliance that merge works correctly/legibly), but happy to change if desired.

Reply:
Ah, never mind, the merging is apparently the recommended (or at least documented way): https://docs.python.org/3/library/subprocess.html#subprocess.check_output

Reply:
As an added benefit, this makes the exception message (& diff) much smaller.  :)

Reply:
Well, the merge was what's causing the json errors... It adds the logs into the output, making the json unparseable, e.g.:
```
Error parsing output of cargo metadata as json; received:
                          b'    Updating crates.io index\n Downloading crates ...\n  Downloaded unindent v0.1.8\n  Downloaded target-lexicon v0.12.3\n  Downloaded pyo3-macros v0.16.5\n  Downloaded cfg-if v1.0.0\n  Downloaded scopeguard v1.1.0\n  Downloaded unicode-xid v0.2.2\n  Downloaded proc-macro2 v1.0.36\n  Downloaded indoc v1.0.4\n  Downloaded instant v0.1.12\n  Downloaded once_cell v1.10.0\n  Downloaded lock_api v0.4.6\n  Downloaded parking_lot v0.11.2\n  Downloaded bitflags v1.3.2\n  Downloaded pyo3-ffi v0.16.5\n  Downloaded quote v1.0.16\n  Downloaded syn v1.0.89\n  Downloaded pyo3-macros-backend v0.16.5\n  Downloaded smallvec v1.8.0\n  Downloaded redox_syscall v0.2.11\n  Downloaded parking_lot_core v0.8.5\n  Downloaded pyo3 v0.16.5\n  Downloaded winapi v0.3.9\n  Downloaded libc v0.2.121\n  Downloaded pyo3-build-config v0.16.5\n  Downloaded winapi-x86_64-pc-windows-gnu v0.4.0\n  Downloaded winapi-i686-pc-windows-gnu v0.4.0\n{"packages":[{"name":"bitflags",
```

Reply:
Ah, I see!

Now that I realise that these updating / downloading messages come through stderr - it's probably quite nice to get them as output by default (i.e. not capture stderr)? 

Could maybe check the `--quiet` flag - if set, capture and include `stderr` in the exception message, otherwise just don't capture stderr? What do you think of that?

I _think_ we're pretty much there now - thanks for the patience / iteration on this!

Reply:
I went back to trying the non-merged version for now. :)

Reply:
This is looking better now. :)